Mathlib/Data/Finset/Sum.lean

```diff
@@ -110,27 +109,20 @@ forms a quasi-inverse to `disjSum`, in that it recovers its left input.
 
 See also `List.partitionMap`.
 -/
-def toLeft (s : Finset (α ⊕ β)) : Finset α :=
-  s.disjiUnion (Sum.elim singleton (fun _ => ∅)) <| by
-    simp [Set.PairwiseDisjoint, Set.Pairwise, Function.onFun, eq_comm]
+def toLeft (u : Finset (α ⊕ β)) : Finset α := (u.subtype _).map Equiv.sumIsLeft.toEmbedding
```

Comment:
This is twice as fast
```suggestion
def toLeft (u : Finset (α ⊕ β)) : Finset α :=
  u.filterMap (Sum.elim some fun _ => none) (by aesop)
```
as it does one pass rather than two. Probably the `aesop` is hiding a lemma that could be written.

Reply:
Moved to #23079, since this change alone compiles.